test/functional/spec-runner/index.js

```diff
@@ -121,7 +121,7 @@ function parseRunOn(runOn) {
 }
 
 function generateTopologyTests(testSuites, testContext, filter) {
-  testSuites.forEach(testSuite => {
+  for (const testSuite of testSuites) {
```

Comment:
Sorry about the code churn here, reworking this use normal loops and async await made debugging far more straight forward. For example, if configuring the fail point fails you find out when stepping over that line vs having to look for the ultimate catch clause that handled it.